fhir-server/src/main/java/com/ibm/fhir/server/util/RestAuditLogger.java

```diff
@@ -368,29 +373,55 @@ public static void logBundle(HttpServletRequest request, Bundle requestBundle, B
                 for (Entry bundleEntry : requestBundle.getEntry()) {
                     if (bundleEntry.getRequest() != null && bundleEntry.getRequest().getMethod() != null) {
                         requestMethod = bundleEntry.getRequest().getMethod();
+                        boolean operation =  bundleEntry.getRequest()
+                                                .getUrl().getValue().contains("$");
                         switch (HTTPVerb.Value.from(requestMethod.getValue())) {
                         case GET:
-                            readCount++;
+                            if (operation) {
+                                executeCount++;
+                            } else {
+                                readCount++;
+                            }
                             break;
                         case POST:
-                            createCount++;
+                            if (operation) {
+                                executeCount++;
+                            } else {
+                                createCount++;
+                            }
                             break;
                         case PUT:
-                            updateCount++;
+                            if (operation) {
+                                executeCount++;
+                            } else {
+                                updateCount++;
+                            }
+                            break;
+                        case DELETE:
+                            if (operation) {
+                                executeCount++;
+                            } else {
+                                deleteCount++;
+                            }
                             break;
                         default:
                             break;
                         }
                     }
                 }
             }
-            entry.getContext().setBatch(Batch.builder().resourcesCreated(createCount).resourcesRead(readCount).resourcesUpdated(updateCount).build());
+            entry.getContext()
+                .setBatch(Batch.builder()
+                    .resourcesCreated(createCount)
+                    .resourcesRead(readCount)
+                    .resourcesUpdated(updateCount)
+                    .resourcesDeleted(deleteCount)
+                    .resourcesExecuted(executeCount)
+                    .build());
             entry.setDescription("FHIR Bundle request");
 
-            // Team discussion results in a recommendation of 'E'
-            // New logic should ensure consistency, all R, all U, all D, all C
-            // when mixed default to E
-            entry.getContext().setAction("E");
+            //
```

Comment:
either remove or summarize the behavior.  i think better to remove and add any pertinent info to the selectActionForBundle javadoc
```suggestion
```

Reply:
I ended up creating the helper method instead.